cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/notifier/SlackSelfHealingNotifier.java

```diff
@@ -44,8 +47,10 @@ public void configure(Map<String, ?> config) {
         _slackIcon = (String) config.get(SLACK_SELF_HEALING_NOTIFIER_ICON);
         _slackChannel = (String) config.get(SLACK_SELF_HEALING_NOTIFIER_CHANNEL);
         _slackUser = (String) config.get(SLACK_SELF_HEALING_NOTIFIER_USER);
+        _slackPreamble = (String) config.get(SLACK_SELF_HEALING_NOTIFIER_PREAMBLE);
         _slackIcon = _slackIcon == null ? DEFAULT_SLACK_SELF_HEALING_NOTIFIER_ICON : _slackIcon;
         _slackUser = _slackUser == null ? DEFAULT_SLACK_SELF_HEALING_NOTIFIER_USER : _slackUser;
+        _slackPreamble = _slackPreamble == null ? DEFAULT_SLACK_SELF_HEALING_NOTIFIER_PREAMBLE : _slackPreamble;
```

Comment:
You can use getOrDefault() method.
config.getOrDefault(SLACK_SELF_HEALING_NOTIFIER_PREAMBLE, DEFAULT_SLACK_SELF_HEALING_NOTIFIER_PREAMBLE)

Reply:
That won't work as the default will have to be a string and what comes out of the config map is an Object. It has to be a two step process, somewhere the object has to be cast/converted to a string. So I just followed what had be done with other optional configs.

Reply:
I could make it one line of course but I am not sure that buys us much?

Reply:
That's fine then. Will approve. Thanks!